### ISSUE TITLE
Add Bootstrap Config to Envoy Deployment

### ISSUE THREAD

**Issue by danehans** [envoyproxy/gateway#201]
https://github.com/envoyproxy/gateway/pull/200 added initial support for deploying Envoy. However, this PR did not include creating Envoy's bootstrap config. A solution should be added for provisioning the bootstrap config as part of the Envoy deployment.

**Comment by danehans** [comment]
@youngnick I see that Contour uses an init container to manage the bootstrap config. What is the reason why a configmap containing the bootstrap config can't simply be mounted to the Envoy container?

**Comment by danehans** [comment]
xref https://github.com/projectcontour/contour/issues/1

**Comment by youngnick** [comment]
I think that there were a couple of reasons:
- We had some information that needed the pod to be up before we could generate the bootstrap (although I can't recall what that is). I think this one is mitigated in the Envoy Gateway case by the fact that we're controlling the whole creation process.
- We wanted not to have any secret stuff in the bootstrap visible in the cluster. I think that there's actually not much that's very secret in the bootstrap config itself - all the secret stuff will be stored in Kubernetes Secrets (basically, the TLS details for connecting to the xDS server in EG).

In summary, I think that using a configmap for EG should be fine, if there are any problems with the approach, we can always change back later.

**Comment by danehans** [comment]
Help wanted to either update take either of the following approaches to manage the bootstrap config:
1. Add the bootstrap config cli arg to the Envoy deployment.
2. Manage a configmap that holds the bootstrap config and mount the configmap to the envoy deployment.
